.github/actions/setup-macos-aarch64-builder/action.yaml

```diff
@@ -45,5 +45,7 @@ runs:
         rustup component add rustfmt
     - name: Setup rust cache
       uses: Swatinem/rust-cache@v2
```

Comment:
TIL that `rust-cache` caches the Rust binaries too

https://github.com/Swatinem/rust-cache?tab=readme-ov-file#cache-details

Reply:
Also, something I noticed was maybe we should pin to an actual hash rather than a tag that is updated

https://github.com/Swatinem/rust-cache/tags

I think security best practice would be to pin to https://github.com/Swatinem/rust-cache/commit/98c8021b550208e191a6a3145459bfc9fb29c4c0 for example rather than https://github.com/Swatinem/rust-cache/releases/tag/v2

However, this wasn't introduced in this PR so I don't think it needs to be fixed here